prql/src/ast.rs

```diff
@@ -118,49 +117,15 @@ pub struct Table {
     pub pipeline: Pipeline,
 }
 
-// We use `NamedArg` for both the FuncCall and the function parameter. They're
-// very similar, so it's fine; though we could split them out if that became
-// helpful.
-#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
-pub struct NamedArg {
-    pub name: Ident,
-    pub arg: Box<Item>,
-}
-
 #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
 pub struct NamedExpr {
-    pub alias: Option<Ident>,
+    pub name: Ident,
     pub expr: Box<Item>,
 }
 
 impl NamedExpr {
-    pub fn unnamed(expr: Item) -> Self {
-        NamedExpr {
-            alias: None,
-            expr: Box::new(expr),
-        }
-    }
-
-    /// Almost all function parameters and list items are wrapped into NamedExpr, which
-    /// can be passed either to this function or to into_list_with_names.
-    pub fn discard_name(self) -> Result<Item> {
-        if self.alias.is_some() {
-            bail!("Cannot use alias for: {self:?}")
-        }
-        Ok(*self.expr)
-    }
-
-    /// Often we don't care whether a List or single item is passed; e.g.
-    /// `select x` vs `select [x, y]`. This equalizes them both to a vec of
-    /// expression, including unnesting any ListItems.
-    pub fn coerce_to_named_exprs(self) -> Vec<NamedExpr> {
-        match self {
-            NamedExpr { alias: None, expr } => match *expr {
-                Item::List(items) => items.into_iter().map(|x| x.into_inner()).collect(),
-                x => vec![NamedExpr::unnamed(x)],
-            },
-            n_expr => vec![n_expr],
-        }
+    pub fn unnamed(expr: Item) -> Item {
```

Comment:
Should this be a method? e.g. `pub fn unnamed(self) -> Item {`

(possibly `into_unnamed` given it's a move, but no stress)

Reply:
I see we also have [`discard_name`](https://github.com/max-sixty/prql/pull/250/commits/69ec7a1d023f5c448b04eb99b029402022d9c6ab#diff-03d9ab37a5c3933dabd1520ee3aacc9f093aad4272ecc9abf7027f148ddeea23R127).

Maybe we don't need this; given `expr` is pub already, and they're equivalent. (but no need to block the merge, we can come back to these later)

Reply:
This is a leftover relic...